openmetadata-service/src/main/java/org/openmetadata/service/resources/elasticSearch/ElasticSearchResource.java

```diff
@@ -0,0 +1,482 @@
+    private volatile int totalFailedCount = 0;
+    private volatile int totalRequests = 0;
+    private String requestIssuer = "anonymous";
+    private String entityFQN;
+    private final CollectionDAO dao;
+    private Long startTime;
+    private Long originalStartTime;
+
+    public BulkProcessorListener(CollectionDAO dao) {
+      this.dao = dao;
+    }
+
+    @Override
+    public void beforeBulk(long executionId, BulkRequest bulkRequest) {
+      int numberOfActions = bulkRequest.numberOfActions();
+      LOG.info("Executing bulk [{}] with {} requests", executionId, numberOfActions);
+    }
+
+    @Override
+    public void afterBulk(long executionId, BulkRequest bulkRequest, BulkResponse bulkResponse) {
+      try {
+        int failedCount = 0;
+        int successCount;
+        FailureDetails failureDetails = null;
+        for (BulkItemResponse bulkItemResponse : bulkResponse) {
+          if (bulkItemResponse.isFailed()) {
+            BulkItemResponse.Failure failure = bulkItemResponse.getFailure();
+            failureDetails = new FailureDetails();
+            failureDetails.setLastFailedAt(
+                Date.from(LocalDateTime.now().atZone(ZoneId.systemDefault()).toInstant()).getTime());
+            failureDetails.setLastFailedReason(
+                String.format("ID [%s]. Reason : %s", failure.getId(), failure.getMessage()));
+            failedCount++;
+          }
+        }
+        successCount = bulkResponse.getItems().length - failedCount;
+        updateFailedAndSuccess(failedCount, successCount);
+
+        System.out.println("TOTAL : " + totalRequests + " SUCCESS : " + successCount + " FAILEWD : " + failedCount);
+        // update stats in DB
+        Long time = Date.from(LocalDateTime.now().atZone(ZoneId.systemDefault()).toInstant()).getTime();
+        EventPublisherJob updateJob =
+            new EventPublisherJob()
+                .withName(String.format("ElasticSearchReindexJob:%s", entityFQN))
+                .withPublisherType(EventPublisherJob.PublisherType.ELASTIC_SEARCH)
+                .withRunMode(EventPublisherJob.RunMode.BATCH)
+                .withStatus(EventPublisherJob.Status.RUNNING)
+                .withTimestamp(time)
+                .withStats(
+                    new Stats().withFailed(totalFailedCount).withSuccess(totalSuccessCount).withTotal(totalRequests))
+                .withStartedBy(requestIssuer)
+                .withFailureDetails(failureDetails)
+                .withStartTime(originalStartTime);
+        if (totalRequests == totalFailedCount + totalSuccessCount) {
+          updateJob.setStatus(EventPublisherJob.Status.SUCCESS);
+          updateJob.setEndTime(time);
+        }
+        dao.entityExtensionTimeSeriesDao()
+            .update(entityFQN, ELASTIC_SEARCH_EXTENSION, JsonUtils.pojoToJson(updateJob), startTime);
+        startTime = time;
+      } catch (IOException e) {
+        throw new RuntimeException(e);
```

Comment:
if there is IOException happened we should set the status in the latest run and record the exception that can be viewable from the UI.
Users should be able to re-trigger.